File: lib/core/ai/models/mira_occasion.dart
/// Occasions supported by Mira styling recommendations.
enum MiraOccasion {
  wedding,
  work,
  casual,
  university,
  evening,
  eid,
  interview;

  String get id => name;

  String get labelAr {
    switch (this) {
      case MiraOccasion.wedding:
        return 'زفاف';
      case MiraOccasion.work:
        return 'عمل';
      case MiraOccasion.casual:
        return 'كاجوال';
      case MiraOccasion.university:
        return 'جامعة';
      case MiraOccasion.evening:
        return 'سهرة';
      case MiraOccasion.eid:
        return 'عيد';
      case MiraOccasion.interview:
        return 'مقابلة';
    }
  }

  String get labelEn {
    switch (this) {
      case MiraOccasion.wedding:
        return 'Wedding';
      case MiraOccasion.work:
        return 'Work';
      case MiraOccasion.casual:
        return 'Casual';
      case MiraOccasion.university:
        return 'University';
      case MiraOccasion.evening:
        return 'Evening';
      case MiraOccasion.eid:
        return 'Eid';
      case MiraOccasion.interview:
        return 'Interview';
    }
  }

  static MiraOccasion? fromId(String? id) {
    if (id == null || id.isEmpty) return null;
    for (final o in MiraOccasion.values) {
      if (o.id == id) return o;
    }
    return null;
  }
}

File: mira-api/src/ai/contracts/mira-occasion.ts
export enum MiraOccasion {
  Wedding = 'wedding',
  Work = 'work',
  Casual = 'casual',
  University = 'university',
  Evening = 'evening',
  Eid = 'eid',
  Interview = 'interview',
}

export const MIRA_OCCASION_LABELS: Record<
  MiraOccasion,
  { ar: string; en: string }
> = {
  [MiraOccasion.Wedding]: { ar: 'زفاف', en: 'Wedding' },
  [MiraOccasion.Work]: { ar: 'عمل', en: 'Work' },
  [MiraOccasion.Casual]: { ar: 'كاجوال', en: 'Casual' },
  [MiraOccasion.University]: { ar: 'جامعة', en: 'University' },
  [MiraOccasion.Evening]: { ar: 'سهرة', en: 'Evening' },
  [MiraOccasion.Eid]: { ar: 'عيد', en: 'Eid' },
  [MiraOccasion.Interview]: { ar: 'مقابلة', en: 'Interview' },
};

const OCCASION_INDEX: Record<MiraOccasion, number> = {
  [MiraOccasion.Wedding]: 0,
  [MiraOccasion.Work]: 1,
  [MiraOccasion.Casual]: 2,
  [MiraOccasion.University]: 3,
  [MiraOccasion.Evening]: 4,
  [MiraOccasion.Eid]: 5,
  [MiraOccasion.Interview]: 6,
};

export function occasionIndex(occasion: MiraOccasion): number {
  return OCCASION_INDEX[occasion];
}

export function parseOccasion(value: string): MiraOccasion | null {
  const normalized = value.trim().toLowerCase();
  return (Object.values(MiraOccasion) as string[]).includes(normalized)
    ? (normalized as MiraOccasion)
    : null;
}
